client/src/Components/Results.js

```diff
@@ -1,23 +1,119 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 
 export default function Results(props) {
+	const [studentNames, setStudentNames]=useState([]);
+	const [selQuizQuestions, setSelQuizQuestions]=useState([]);
+	const [quizRoute,setQuizRoute]=useState("");
+	const [allResults, setAllResults]=useState(null);
+	const [quizSelected, setQuizSelected]=useState(null);
+	const fetchResults =()=>{
+		fetch(`http://localhost:3100/api/results/${quizRoute}`)
+			.then((response)=>response.json())
+			.then((data)=>{
+				setAllResults(data);
+			});
+	};
 
-	if(props.quizes){
+	const findQuestionResult=(actualQuestionId, oneName)=>{
+		let resultsforThisQuestion = allResults.filter((el)=>el.question_id==actualQuestionId);
+
+		let studentsAttemptsOnQuestion = resultsforThisQuestion.filter((el)=>el.studentName==oneName);
+		if(studentsAttemptsOnQuestion.length==0){
+			return("unknown");
+		}
+		if(studentsAttemptsOnQuestion.length>0){
+			console.log({ studentsAttemptsOnQuestion });
+		}
+
+		if(studentsAttemptsOnQuestion!==undefined&&studentsAttemptsOnQuestion[0].correct){
+			return("corect");
+		} else{
+			return("incorect");
+		}
+	};
+	const findStudenAnswerResult=(actualQuestionId, oneName)=>{
+		let studentAnsweredId = allResults.filter((el)=>el.question_id==actualQuestionId);
+		let studentAnsweredQuestion = studentAnsweredId.filter((el)=>el.studentName==oneName);
+		if(studentAnsweredQuestion.length==0){
+			return("unknown");
+		}
+		if(studentAnsweredQuestion.length>0){
+		}
+		if(studentAnsweredQuestion!==undefined&&studentAnsweredQuestion[0]){
+			return(studentAnsweredQuestion[0].value);
+		}
+	};
+	useEffect(() => {
+		fetchResults();
+	},[quizRoute]);
```

Comment:
ok you call the function in use effect, why you didn't do it directly inside the useEffect?